Modules/Filtering/ImageGrid/include/itkResampleImageFilter.h

```diff
@@ -111,16 +111,16 @@ class ITK_TEMPLATE_EXPORT ResampleImageFilter : public ImageToImageFilter<TInput
   itkTypeMacro(ResampleImageFilter, ImageToImageFilter);
 
   /** Number of dimensions. */
-  static constexpr unsigned int ImageDimension = TOutputImage::ImageDimension;
```

Comment:
Does an ImageDimension value need to persist for compatibility?

Reply:
Yes it could break compatibility, thanks for catching this. I have created another variable called `ImageDimension` for now.. but `ImageDimension` is very Confusing if we have different Input and output dimensions. Unfortunately, `ImageDimension` (OutputImageDimension) is (was) used in many other places in this filter. 
IMHO this variable should be renamed to `OutputImageDimension` inspite of this PR but that is a different issue.

Reply:
We should have `ImageDimension` if `ITK_LEGACY_REMOVE` is OFF. Example of use
https://github.com/InsightSoftwareConsortium/ITK/blob/21fc5c94b58c00f5db1a8ff4d2a18e2fbeb59014/Modules/Core/Common/include/itkLoggerBase.h#L93-L98

Reply:
Thanks.